### PR TITLE
chore: move trigger methods to its own package `react-trigger`

### PR DESCRIPTION
**Pull request by bsunderhus**
Main reason for this split would only be better separation of concern. The way I see trigger has its own domain logic meanwhile `react-utilities` should be more devoted for general purpose things. I have the same feeling for slots (it should be separated from utilities), but I guess its already too late for that...

1. Moves trigger methods from `@fluentui/react-utilities` to `@fluentui/react-trigger`
2.  Updates dependencies from `@fluentui/react-utilities` to `@fluentui/react-trigger`
    - `react-dialog`
    - `react-menu`
    - `react-overflow`
    - `react-popover`
    - `react-tooltip`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                                                                    |          `533 B`<br />`341 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr>                                                    |          `554 B`<br />`348 B` |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |   `78.914 kB`<br />`24.06 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `83.511 kB`<br />`20.921 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.381 kB`<br />`13.696 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `14.95 kB`<br />`5.989 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `68.349 kB`<br />`19.039 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |     `22.6 kB`<br />`7.205 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `23.503 kB`<br />`7.497 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |    `24.05 kB`<br />`7.067 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `36.119 kB`<br />`9.647 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |   `43.144 kB`<br />`10.86 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `38.813 kB`<br />`10.551 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `46.228 kB`<br />`11.933 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `51.888 kB`<br />`11.127 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `67.002 kB`<br />`19.261 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `62.684 kB`<br />`18.177 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `8.561 kB`<br />`3.601 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |     `9.604 kB`<br />`3.94 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |    `8.662 kB`<br />`3.656 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `74.636 kB`<br />`24.186 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `74.236 kB`<br />`24.086 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> |  `188.689 kB`<br />`52.37 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.394 kB`<br />`11.007 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `82.755 kB`<br />`24.581 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `16.459 kB`<br />`5.902 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |    `10.78 kB`<br />`4.264 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `23.757 kB`<br />`7.704 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |     `9.338 kB`<br />`3.86 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `11.784 kB`<br />`4.867 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `116.589 kB`<br />`35.777 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `119.658 kB`<br />`36.296 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `10.685 kB`<br />`4.104 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `102.963 kB`<br />`31.553 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `10.576 kB`<br />`3.875 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.851 kB`<br />`1.964 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |     `19.7 kB`<br />`7.404 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `15.755 kB`<br />`5.883 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |   `35.56 kB`<br />`11.929 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |     `14.248 kB`<br />`5.7 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `20.846 kB`<br />`7.346 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `31.526 kB`<br />`10.046 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `44.102 kB`<br />`12.425 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `19.977 kB`<br />`6.438 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `32.097 kB`<br />`10.27 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `11.782 kB`<br />`4.605 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `15.092 kB`<br />`5.044 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `25.013 kB`<br />`8.133 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `41.535 kB`<br />`14.639 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                                                                       |          `180 B`<br />`159 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e3e67958e604ceb8b36e1954ba595a0f50a196fd'>e3e67958e604ceb8b36e1954ba595a0f50a196fd</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/Avatar-mount_base.html">1263</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/Avatar-mount.html">1272</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/Button-mount_base.html">938</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/Button-mount.html">932</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProvider-mount_base.html">1536</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProvider-mount.html">1488</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-mount_base.html">574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-mount.html">585</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">544</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">577</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">563</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/MakeStyles-mount_base.html">1947</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/MakeStyles-mount.html">1947</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/SpinButton-mount_base.html">2315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24923/perf-test-react-components/SpinButton-mount.html">2306</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ling1726 — DISMISSED** [review]

File: packages/react-components/react-dialog/package.json
Comment (ling1726) [inline packages/react-components/react-dialog/package.json:38]: this should not be an alpha package, especially since we moved it out of a stable package

File: change/@fluentui-react-trigger-878d2c89-74e6-475d-bd81-c43652fb8f57.json
Comment (ling1726) [inline change/@fluentui-react-trigger-878d2c89-74e6-475d-bd81-c43652fb8f57.json:2]:
```suggestion
  "type": "patch",
```

you should revert the manual bump you did and just change the change type in the change file. Beachball will automatically bump it to the next available non-prerelease version

This is why I'm not a fan of magic tools like beachball :D

File: packages/react-components/react-trigger/package.json
Comment (ling1726) [inline packages/react-components/react-trigger/package.json:3]:
```suggestion
  "version": "9.0.0-alpha.0",
```

Comment (ling1726) [reply packages/react-components/react-trigger/package.json:3]: Please also update `disallowedChangeTypes`

**Review by ling1726 — APPROVED** [review]

**Review by behowell — COMMENTED** [review]
This change looks good technically. However, the description doesn't say why it's necessary to move this into a separate package. I'm worried about excessive proliferation of packages if we start moving each utility to its own package. Could you update the description with more details about what problem this is solving?

If you'd like, you can set this PR to auto-merge after updating the description, so it will merge once I see the update and approve.

Thanks!

**Review by behowell — CHANGES_REQUESTED** [review]
So it sounds like this is refactoring; not solving a technical problem, right? Before making a new package for this, it would be good to have a plan for how we decide when a utility should live in react-utilities vs. its own package. That way we can have a consistent and predictable set of packages. 

If the goal is to keep related utilities together, then perhaps they should just be in a sub-folder of react-utilities, instead of their own package.

Or, if the goal is to separate out the fluentui-isms like triggers and slots, out from general-purpose utilities, then perhaps we name this package `react-core` or something, and move the slots and other core functionality into it.

Would it be possible to write a (short?) RFC about how that would work, and discuss it with the team at the next tech sync?

**Comment by bsunderhus** [comment]
> So it sounds like this is refactoring; not solving a technical problem, right? 

Yes, this is definitely just a refactoring, and a simple one.

> If the goal is to keep related utilities together, then perhaps they should just be in a sub-folder of react-utilities, instead of their own package.

> Or, if the goal is to separate out the fluentui-isms like triggers and slots, out from general-purpose utilities, then perhaps we name this package react-core or something, and move the slots and other core functionality into it.

The problem is not just splitting from general-purpose utilities methods, its more about the priority those domain specific utilities require in our own library.

The way I see utilities provided on the `hooks` are simple generic use case hooks that can be easily documented with comments. On the case of both slots and triggers _utilities_ on the other hand, they have their own domain and its very specific use cases. Demanding proper documentation and examples on our side to ensure proper usage, simply adding comments doesn't get this done.

I believe a `react-core` package would end up being equivalent to our current approach of dropping everything related in the same folder. By splitting those domain specific cases we're giving them the actual relevance they deserve to ensure we'll have them properly documented (which is not necessarily the case at the moment), as an undocumented package is way more critical than an undocumented folder.

**Comment by bsunderhus** [comment]
After internal discussion, I think it's a majority decision to keep the current folder structure the way it is, on the addendum of a commitment on our side for better documentation on those domain specific folders.

**Pull request closed without merging**
